src/Generating/FinishGen.cpp

```diff
@@ -776,36 +837,26 @@ void cFinishGenSprinkleFoliage::GenFinish(cChunkDesc & a_ChunkDesc)
 					{
 						a_ChunkDesc.SetBlockTypeMeta(x, ++Top, z, E_BLOCK_TALL_GRASS, E_META_TALL_GRASS_GRASS);
 					}
-					else if (TryAddSugarcane(a_ChunkDesc, x, Top, z))
-					{
-						++Top;
-					}
 					else if ((val1 > 0.5) && (val2 < -0.5))
 					{
 						a_ChunkDesc.SetBlockTypeMeta(x, ++Top, z, E_BLOCK_PUMPKIN, static_cast<int>(val3 * 8) % 4);
 					}
+					else
+					{
+						TryAddSugarcane(a_ChunkDesc, x, Top, z);
+					}
```

Comment:
I had changed it because updating the `Top` variable was now useless (as it was done inside the `TryAddSugarcane` function), but indeed, this could be one of the reasons of its rarity. I'll change it.